### PR TITLE
Moving networkfirewall(2), networkmanager(9) resources to v1beta1 version

### PR DESCRIPTION
**Pull request by svscheg**
### Description of your changes
Moving `networkfirewall` (2) resources to v1beta1 version
- [x] [`aws_networkfirewall_firewall_policy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_firewall_policy)
- [x] [`aws_networkfirewall_rule_group`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_rule_group)

Moving `networkmanager` (9) resources to v1beta1 version
- [x] [`aws_networkmanager_connection`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_connection)
- [x] [`aws_networkmanager_customer_gateway_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_customer_gateway_association)
- [x] [`aws_networkmanager_device`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_device)
- [x] [`aws_networkmanager_global_network`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_global_network)
- [x] [`aws_networkmanager_link`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_link)
- [x] [`aws_networkmanager_link_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_link_association)
- [x] [`aws_networkmanager_site`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_site)
- [x] [`aws_networkmanager_transit_gateway_connect_peer_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_transit_gateway_connect_peer_association)
- [x] [`aws_networkmanager_transit_gateway_registration`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_transit_gateway_registration)

Fixes https://github.com/upbound/provider-aws/issues/378

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Manually
Uptests
Moving `networkfirewall` (2) resources to v1beta1 version
- [x] [`aws_networkfirewall_firewall_policy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_firewall_policy) - https://github.com/upbound/provider-aws/actions/runs/3795820981/jobs/6455322608
- [x] [`aws_networkfirewall_rule_group`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_rule_group) - https://github.com/upbound/provider-aws/actions/runs/3795822120/jobs/6455324880

Moving `networkmanager` (9) resources to v1beta1 version
- [x] [`aws_networkmanager_connection`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_connection) - https://github.com/upbound/provider-aws/actions/runs/3795823066
- [x] [`aws_networkmanager_customer_gateway_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_customer_gateway_association) - https://github.com/upbound/provider-aws/actions/runs/3795824069/jobs/6455328676
- [x] [`aws_networkmanager_device`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_device) - https://github.com/upbound/provider-aws/actions/runs/3795825206
- [x] [`aws_networkmanager_global_network`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_global_network) - https://github.com/upbound/provider-aws/actions/runs/3795826586/jobs/6455333504
- [x] [`aws_networkmanager_link`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_link) - https://github.com/upbound/provider-aws/actions/runs/3795827396/jobs/6455335504
- [x] [`aws_networkmanager_link_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_link_association) - https://github.com/upbound/provider-aws/actions/runs/3795835226/jobs/6455351312
- [x] [`aws_networkmanager_site`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_site) - https://github.com/upbound/provider-aws/actions/runs/3795836311/jobs/6455353058
- [x] [`aws_networkmanager_transit_gateway_connect_peer_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_transit_gateway_connect_peer_association) - https://github.com/upbound/provider-aws/actions/runs/3795837039/jobs/6455354432
- [x] [`aws_networkmanager_transit_gateway_registration`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkmanager_transit_gateway_registration) - https://github.com/upbound/provider-aws/actions/runs/3795837743/jobs/6455355767

### CODE REVIEW

**Comment by svscheg** [comment]
/test-examples="examples/networkfirewall/firewallpolicy.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/networkfirewall/rulegroup.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/networkmanager/connection.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/networkmanager/customergatewayassociation.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/networkmanager/device.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/networkmanager/globalnetwork.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/networkmanager/link.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/networkmanager/linkassociation.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/networkmanager/site.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/networkmanager/transitgatewayconnectpeerassociation.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/networkmanager/transitgatewayregistration.yaml"

File: config/externalname.go [hunk 1]
```diff
@@ -1956,6 +1956,48 @@ var ExternalNameConfigs = map[string]config.ExternalName{
...
+	// Example: arn:aws:network-firewall:us-west-1:123456789012:stateful-rulegroup/example
+	"aws_networkfirewall_rule_group": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:stateful-rulegroup/{{ .external_name }}"),
+
+	// networkmanager
+	//
+	// aws_networkmanager_global_network can be imported using the global network ID
+	"aws_networkmanager_global_network": config.IdentifierFromProvider,
+	// aws_networkmanager_site can be imported using the site ARN
+	// Example: arn:aws:networkmanager::123456789012:site/global-network-0d47f6t230mz46dy4/site-444555aaabbb11223
+	// TODO: I am not sure that, the last element of the template string will provide this string: site-444555aaabbb11223. If not, then change configuration to IdentifierFromProvider
```
Comment (sergenyalcin) [inline config/externalname.go:1975]: We may consider removing this TODO statement.

File: config/externalname.go [hunk 2]
```diff
@@ -1956,6 +1956,48 @@ var ExternalNameConfigs = map[string]config.ExternalName{
...
+	"aws_networkmanager_site": config.IdentifierFromProvider,
+	// aws_networkmanager_link can be imported using the link ARN
+	// Example: arn:aws:networkmanager::123456789012:link/global-network-0d47f6t230mz46dy4/link-444555aaabbb11223
+	// TODO: I am not sure that, the last element of the template string will provide this string: link-444555aaabbb11223. If not, then change configuration to IdentifierFromProvider
```
Comment (sergenyalcin) [inline config/externalname.go:1979]: We may consider removing this TODO statement.

File: config/externalname.go [hunk 3]
```diff
@@ -1956,6 +1956,48 @@ var ExternalNameConfigs = map[string]config.ExternalName{
...
+	"aws_networkmanager_link": config.IdentifierFromProvider,
+	// aws_networkmanager_link_association can be imported using the global network ID, link ID and device ID
+	// Example: global-network-0d47f6t230mz46dy4,link-444555aaabbb11223,device-07f6fd08867abc123
+	"aws_networkmanager_link_association": config.TemplatedStringAsIdentifier("", "{{ .parameters.global_network_id }},{{ .parameters.link_id }},{{ .parameters.device_id }}"),
+	// aws_networkmanager_device can be imported using the device ARN
+	// Example: arn:aws:networkmanager::123456789012:device/global-network-0d47f6t230mz46dy4/device-07f6fd08867abc123
+	// TODO: I am not sure that, the last element of the template string will provide this string: device-07f6fd08867abc123. If not, then change configuration to IdentifierFromProvider
```
Comment (sergenyalcin) [inline config/externalname.go:1986]: We may consider removing this TODO statement.

File: config/externalname.go [hunk 4]
```diff
@@ -1956,6 +1956,48 @@ var ExternalNameConfigs = map[string]config.ExternalName{
...
+	"aws_networkmanager_device": config.IdentifierFromProvider,
+	// aws_networkmanager_connection can be imported using the connection ARN
+	// Example: arn:aws:networkmanager::123456789012:device/global-network-0d47f6t230mz46dy4/connection-07f6fd08867abc123
+	// TODO: I am not sure that, the last element of the template string will provide this string: connection-07f6fd08867abc123. If not, then change configuration to IdentifierFromProvider
```
Comment (sergenyalcin) [inline config/externalname.go:1990]: We may consider removing this TODO statement.

**Review by sergenyalcin — COMMENTED** [review]
Thanks @svscheg I left a few comments.

**Review by sergenyalcin — APPROVED** [review]
Thanks @svscheg LGTM!

**Pull request merged**